File: src/chipschallenge/blockreactions/CreatureTo.java
package chipschallenge.blockreactions;

import chipschallenge.Block;

/**
 *
 * @author patrik
 */
public class CreatureTo extends BlockReaction {

    private CreatureTo() {
    }
    private static CreatureTo mInstance = null;

    public static synchronized CreatureTo getInstance() {
        if (mInstance == null) {
            mInstance = new CreatureTo();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        if (moving != standing) {
            die("Ooops! Look out for creatures!");
        }
    }

    public boolean canMove(Block moving, Block standing) {
        return moving.isChip();
    }
}

File: ChipsChallenge.Shared/Blockreactions/CreatureTo.cs
﻿namespace ChipsChallenge.Shared.Blockreactions
{
    public class CreatureTo : NoSlipReaction
    {
        private static CreatureTo instance;

        private CreatureTo()
        {
        }

        public static CreatureTo Instance
        {
            get
            {
                lock (typeof(CreatureTo))
                {
                    return instance ?? (instance = new CreatureTo());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            if (moving != standing)
            {
                Die("Ooops! Look out for creatures!");
            }
        }

        public override bool canMove(Block moving, Block standing)
        {
            return moving.Chip;
        }
    }
}